### PR TITLE
unsafe sysctls: demote its conformance test to e2e test

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind bug
/sig node 

#### What this PR does / why we need it:

> kernel.shm_rmid_forced is in the safe set.
> The current two test cases code are totally the same for Sysctl, test sysctls and Sysctl, allow specified unsafe sysctls. 

#### Which issue(s) this PR fixes:

follow up of https://github.com/kubernetes/kubernetes/pull/99734#discussion_r600864919

#### Special notes for your reviewer:
correct the conformance test case

#### Does this PR introduce a user-facing change?

```release-note
None
```

/cc derekwaynecarr wgahnagl  ehashman

### CODE REVIEW

File: test/e2e/common/node/sysctl.go
Comment (ehashman) [inline test/e2e/common/node/sysctl.go:142]: This test is supposed to be checking if allowed unsafe sysctls work, but now we're showing that it's not working. This seems incorrect to me, at least without an update to the test name and metadata.

**Review by ehashman — CHANGES_REQUESTED** [review]

File: test/e2e/common/node/sysctl.go
Comment (pacoxu) [reply test/e2e/common/node/sysctl.go:142]: To make it work, kubelet should be configured with `--allow-unsafe-sysctls`, and this should not be in a conformance test.

So I will update the test name and metadata then.

File: test/e2e/common/node/sysctl.go
Comment (oomichi) [inline test/e2e/common/node/sysctl.go:121]: If changing the test purpose like this because of duplicated other test, it would be safer to drop this test from conformance test set.
Then we can check the test stability and make it promoted again, I feel.

/cc @oomichi

File: test/e2e/common/node/sysctl.go
Comment (oomichi) [inline test/e2e/common/node/sysctl.go:127]: Don't we need to update this message instead of kernel.shm_rmid_forced sysctl?

**Review by oomichi — COMMENTED** [review]
I missed one more point on my previous review, sorry about that.

File: test/e2e/common/node/sysctl.go
Comment (pacoxu) [reply test/e2e/common/node/sysctl.go]: My careless. Thanks.

Updated.

**Review by oomichi — COMMENTED** [review]
Thanks for updating.

/lgtm

**Comment by pacoxu** [comment]
/assign @derekwaynecarr @smarterclayton
for approval

**Comment by pacoxu** [comment]
/assign dims BenTheElder
before code freeze for approval

**Comment by dims** [comment]
@pacoxu you will need sign off from sig-node folks ( from @derekwaynecarr ) and then approval from @smarterclayton 

/milestone v1.22

File: test/e2e/common/node/sysctl.go
Comment (SergeyKanzhelev) [inline test/e2e/common/node/sysctl.go:115]:
```suggestion
	ginkgo.It("should forbid unsafe sysctls [MinimumKubeletVersion:1.21]", func() {
```

File: test/e2e/common/node/sysctl.go
Comment (SergeyKanzhelev) [inline test/e2e/common/node/sysctl.go:133]: unrelated: this comment goes against this issue: https://github.com/kubernetes/kubernetes/issues/71646 I wonder if this can be fixed somehow

Comment (pacoxu) [reply test/e2e/common/node/sysctl.go:133]: Good Point. However, I have not seen related flakes with this test case yet.

We may check the CI flakes continuously.

Comment (liggitt) [reply test/e2e/common/node/sysctl.go:133]: conformance tests should not rely on event presence or content

Comment (pacoxu) [reply test/e2e/common/node/sysctl.go:133]: So we should demote this test to the general e2e test as it should not be a conformance test case.

File: test/e2e/common/node/sysctl.go
Comment (pacoxu) [reply test/e2e/common/node/sysctl.go]: Fixed.

File: test/e2e/common/node/sysctl.go
Comment (pacoxu) [reply test/e2e/common/node/sysctl.go:133]: I may realize the problem here after reading https://github.com/kubernetes/kubernetes/issues/71646. Let me fix it today or tommorrow.

**Comment by pacoxu** [comment]
#71574 seems to be a good example.
Waiting for a pod/container status change may be the solution here.
/hold 
I will update both test cases here later.

File: test/e2e/common/node/sysctl.go
Comment (pacoxu) [inline test/e2e/common/node/sysctl.go:179]: - case 1: should forbid unsafe sysctls
- case 2: should not launch unsafe, but not explicitly enabled sysctls on the node 

It may be a little dup.

File: test/e2e/common/node/sysctl.go
Comment (ehashman) [inline test/e2e/common/node/sysctl.go:158]: Can we keep the comments even if it's not conformance?

File: test/e2e/common/node/sysctl.go
Comment (ehashman) [reply test/e2e/common/node/sysctl.go:179]: Don't we want to still keep an e2e that explicitly permitted, unsafe sysctls are working?

**Comment by ehashman** [comment]
/milestone clear

from SIG Node CI subgroup today, this isn't release-critical

We will split out the bit that removes this test from the conformance suite for 1.22 in a separate PR.

File: test/e2e/common/node/sysctl.go
Comment (pacoxu) [reply test/e2e/common/node/sysctl.go:158]: OK, I add the comments back in #103906.

**Comment by pacoxu** [comment]
> Don't we want to still keep an e2e that explicitly permitted, unsafe sysctls are working?

We should have one, but it should be skipped if there is no node that allows unsafe-sysctls. 
It is a little hard to add that case.

**Pull request closed without merging**
